### DISCUSSION TITLE
string.format fails? with strings in map

### DISCUSSION THREAD

**Discussion by xostia** [arendst/Tasmota#20795] [Support request]
I think that I found a bug in string.format in Berry (Tasmota 13.1), using Maps.

readingsMap is a map and the fields are strings

The values are "2.2A" and "3.0B"

a = readingsMap[VERSIONFW_LABEL]
b = readingsMap[VERSIONSC_LABEL]
string.format ("%s%s",a,b)

I get:

2.2A3.0B

If I write:
string.format("%s %s",readingsMap[VERSIONFW_LABEL],readingsMap[VERSIONSC_LABEL])

I get:

2.2A {'Humidity': 45.1, 'Batlevel': 78.13 ......

... ¡I get the whole map in the string!

It seems to work ok if you only use one Map field in the string.format, but if have more elements (map or other types) it fails this way.

Am I doing something wrong or is it a bug?

[EDIT]

doing the conversion to string (they are strings, tested with type() ), it fails but different:

string.format("%s %s",str(readingsMap[VERSIONFW_LABEL]),str(readingsMap[VERSIONSC_LABEL]))

2.2A2.2A

It takes the first string many times... 

I also tested with other types in the Map, like float or int, string.format fails the same say

Thank you

**Comment by s-hadinger** [comment]
I think it was fixed in https://github.com/arendst/Tasmota/pull/20059

Please try 13.4

**Reply by xostia** [reply]
OK, thank you ... but I tried and my LVGL code needs a lot of work :-(  many changes in LVGL version

using readingsMap.find(VERSIONFW_LABEL) all seem to work.

**Reply by s-hadinger** [reply]
In the interim you can use 13.3 which contains the fix but is still using lvgl 8
